### PR TITLE
Make versions of some nested class/record reduction tests to avoid using nested fields

### PR DESCRIPTION
**Pull request by DanilaFe**
Originally, the tests were written to lock down the fact that `outerclass.nestedclass` was not a valid reduction; now, however, they lock down the fact that the `nestedclass` cannot access the fields of `outerclass`. This PR adds tests that lock down the original behavior (which still doesn't work, it seems).

I also updated the affected tests to have a proper use of the user-defined reductions API, which seems to have changed since.

Reviewed by @lydia-duncan -- thanks!

## Testing
- [x] Updated tests still pass locally

### CODE REVIEW

File: test/users/ferguson/histogram/reductionex_class.future [hunk 1]
```diff
@@ -1,7 +1,3 @@
-bug: nested type specifier does not work for reduction
+language: do we want support for referring to enclosing types' fields?
```
Comment (lydia-duncan) [inline test/users/ferguson/histogram/reductionex_class.future:1]: Language is one of the "area" tags for the repo, but we want to be using the "type" tags for future categorization.  In this case, I think "design" is appropriate.

File: test/users/ferguson/histogram/reductionex_record.future [hunk 2]
```diff
@@ -1,8 +1,3 @@
-bug: nested type specifier does not work for reduction
+language: do we want support for referring to enclosing types' fields?
```
Comment (lydia-duncan) [inline test/users/ferguson/histogram/reductionex_record.future:1]: Ditto here about the future tag

**Review by lydia-duncan — COMMENTED** [review]
Thanks!  You just want to update the .futures so that they use our typically future style.

**Comment by DanilaFe** [comment]
Replaced 'language' with 'design' as requested :)

**Pull request merged**